website/blog/2022-01-14-change-data-capture-with-debezium-and-apache-hudi.md

```diff
@@ -0,0 +1,186 @@
+COPY --from=cp /usr/share/confluent-hub-components/confluentinc-kafka-connect-avro-converter/lib /opt/kafka/plugins/avro/
+USER 1001
+```
+
+Once the Strimzi operator and the Kafka connect are deployed, we can start the Debezium connector.
+
+```
+curl -X POST -H "Content-Type:application/json" -d @connect-source.json [http://localhost:8083/connectors/](http://localhost:8083/connectors/)
+```
+
+The following is an example of a configuration to setup Debezium connector for generating the changelogs for two tables, table1, and table2.
+
+Contents of connect-source.json:
+```json
+{
+  "name": "postgres-debezium-connector",
+  "config": {
+    "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+    "database.hostname": "localhost",
+    "database.port": "5432",
+    "database.user": "postgres",
+    "database.password": "postgres",
+    "database.dbname": "database",
+    "[plugin.name](http://plugin.name)": "pgoutput",
+    "[database.server.name](http://database.server.name)": "postgres",
+    "table.include.list": "schema1.table1,schema1.table2",
+    "publication.autocreate.mode": "filtered",
+    "tombstones.on.delete":"false",
+    "key.converter": "io.confluent.connect.avro.AvroConverter",
+    "key.converter.schema.registry.url": "<schema_registry_host>",
+    "value.converter": "io.confluent.connect.avro.AvroConverter",
+    "value.converter.schema.registry.url": "<schema_registry_host>",
+    "[slot.name](http://slot.name)": "pgslot"
+  }
+}
+```
+
+### Hudi Deltastreamer
+
+Next, we run the Hudi Deltastreamer using spark that will ingest the Debezium changelogs from kafka and write them as a Hudi table. One such instance of the command is shown below that works for Postgres database.  A few key configurations are as follows:
+
+1.  Set the source class to PostgresDebeziumSource.
+2.  Set the payload class to PostgresDebeziumAvroPayload.
+3.  Configure the schema registry URLs for Debezium Source and Kafka Source.
+4.  Set the record key(s) as the primary key(s) of the database table.
+5.  Set the source ordering field (dedup) to _event_lsn
+
+```scala
+spark-submit \\
+  \--jars "/home/hadoop/hudi-utilities-bundle_2.12-0.10.0.jar,/usr/lib/spark/external/lib/spark-avro.jar" \\
+  \--master yarn --deploy-mode client \\
+  \--class org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer /home/hadoop/hudi-packages/hudi-utilities-bundle_2.12-0.10.0-SNAPSHOT.jar \\
+  \--table-type COPY_ON_WRITE --op UPSERT \\
+  \--target-base-path s3://bucket_name/path/for/hudi_table1 \\
+  \--target-table hudi_table1  --continuous \\
+  \--min-sync-interval-seconds 60 \\
+  \--source-class org.apache.hudi.utilities.sources.debezium.PostgresDebeziumSource \\
+  \--source-ordering-field _event_lsn \\
+  \--payload-class org.apache.hudi.common.model.debezium.PostgresDebeziumAvroPayload \\
+  \--hoodie-conf schema.registry.url=[https://localhost:8081](https://localhost/) \\
```

Comment:
In this code block, we need to remove the "\" delimiter at the start of each line.
Also, we shld remove the links for localhost. i.e., [https://localhost:8081](https://localhost/) -> https://localhost:8081. Lines 167 and 168

Reply:
Good catches! I just updated